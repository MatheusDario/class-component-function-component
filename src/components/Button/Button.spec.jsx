const { render, screen, fireEvent } = require('@testing-library/react');
const { Button } = require('.');

describe('<Button />', () => {
  it('should render with a text "load more posts"', () => {
    render(<Button text="load more posts" />);
    const button = screen.getByRole('button', { name: 'load more posts' });
    expect(button).toBeInTheDocument();
  });

  it('should call function on button click', () => {
    const fn = jest.fn();
    render(<Button onClick={fn} text="load more posts" />);
    const button = screen.getByRole('button', { name: 'load more posts' });
    //userEvent.click(button);
    fireEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('should desabled button when disabled for true ', () => {
    render(<Button text="load more posts" disabled />);
    const button = screen.getByRole('button', { name: 'load more posts' });
    expect(button).toBeDisabled();
  });

  it('should enabled when disabled for false ', () => {
    render(<Button text="load more posts" disabled={false} />);
    const button = screen.getByRole('button', { name: 'load more posts' });
    expect(button).toBeEnabled();
  });

  it('should match snap shot ', () => {
    const fn = jest.fn();
    const { container } = render(
      <Button text="Load more" disabled={false} onClick={fn} />,
    );
    expect(container).toMatchSnapshot();
  });
});
