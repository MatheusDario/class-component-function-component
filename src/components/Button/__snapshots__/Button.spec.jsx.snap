// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snap shot  1`] = `
.c0 {
  min-height: 60px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.c1 {
  width: 50%;
  padding: 15px;
  cursor: pointer;
  margin-bottom: 50px;
  transition: all 300ms ease-in-out;
  border-radius: 10px;
  font-size: 1.2rem;
  text-transform: uppercase;
}

.c1:hover {
  background-color: aquamarine;
}

.c1:disabled {
  background-color: gray;
  cursor: not-allowed;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
    >
      Load more
    </button>
  </div>
</div>
`;
